service/resource/deployment/resource.go

```diff
@@ -187,7 +190,85 @@ func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desire
 }
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
-	return nil, nil, nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	currentDeployments, err := toDeployments(currentState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	desiredDeployments, err := toDeployments(desiredState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+
+	var deploymentsToCreate interface{}
+	{
```

Comment:
I've seen a lot of these inner blocks, and in some cases i think i understand that some variables only used inside them are out of scope when they are closed, but what is the benefit of using it here?

Reply:
For me this is for visual blocks and separation. It also often helps with naming because the scope prevents certain collisions.